acceptance_tests/features/environment.py

```diff
@@ -47,38 +37,54 @@ def after_step(context, step):
-    collection_exercise_controller.execute_collection_exercise(survey_id='cb8accda-6118-4d3b-85a3-149e28960c54',
-                                                               period='201812')
-    logger.info('Executing collection exercise', survey_id='02b9c366-7397-42f7-942a-76dc5876d86d', period='1809')
-    collection_exercise_controller.execute_collection_exercise(survey_id='02b9c366-7397-42f7-942a-76dc5876d86d',
-                                                               period='1809')
-    logger.info('Waiting for collection exercises execution process to finish',
-                survey_id='cb8accda-6118-4d3b-85a3-149e28960c54', period='201801')
+    logger.info('Executing collection exercises')
+    # Bricks
+    execute_collection_exercise('cb8accda-6118-4d3b-85a3-149e28960c54', '201801')
+    execute_collection_exercise('cb8accda-6118-4d3b-85a3-149e28960c54', '201812')
+    # QBS
+    execute_collection_exercise('02b9c366-7397-42f7-942a-76dc5876d86d', '1809', ci_type='eQ')
+
+    logger.info('Waiting for collection exercises to finish executing')
     poll_database_for_iac(survey_id='cb8accda-6118-4d3b-85a3-149e28960c54', period='201801')
-    logger.info('Waiting for collection exercises execution process to finish',
-                survey_id='cb8accda-6118-4d3b-85a3-149e28960c54', period='201812')
     poll_database_for_iac(survey_id='cb8accda-6118-4d3b-85a3-149e28960c54', period='201812')
-    logger.info('Waiting for collection exercises execution process to finish',
-                survey_id='02b9c366-7397-42f7-942a-76dc5876d86d', period='1809')
     poll_database_for_iac(survey_id='02b9c366-7397-42f7-942a-76dc5876d86d', period='1809')
+    logger.info('Collection exercises have finished executing')
+
+
+def execute_collection_exercise(survey_id, period, ci_type='SEFT'):
+    logger.info('Executing collection exercise', survey_id=survey_id, period=period, ci_type=ci_type)
+    collection_exercise = collection_exercise_controller.get_collection_exercise(survey_id, period)
+
+    if ci_type == 'eQ':
+        collection_instruments = get_collection_instruments_by_classifier(survey_id, form_type='0001')
+        for collection_instrument in collection_instruments:
+            link_collection_instrument_to_exercise(collection_instrument['id'], collection_exercise['id'])
+    else:
+        upload_seft_collection_instrument(collection_exercise['id'],
+                                          'resources/collection_instrument_files/064_201803_0001.xlsx',
+                                          form_type='0001')
+
+    sample_summary = sample_controller.upload_sample(collection_exercise['id'],
+                                                     'resources/sample_files/business-survey-sample-date.csv')
+    collection_exercise_controller.link_sample_summary_to_collection_exercise(collection_exercise['id'],
+                                                                              sample_summary['id'])
+    time.sleep(5)
```

Comment:
Can we change this time.sleep to a poll mechanism? These sort of poll checks should help with the flakiness of the tests

Reply:
I'll have a look into it but might not be changed with this PR. There's quite a lot getting changed already